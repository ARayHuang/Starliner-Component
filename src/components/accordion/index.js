import React from 'react';
import PropTypes from 'prop-types';
import {
	Accordion as CAccordion,
	AccordionItem,
	AccordionButton,
	AccordionPanel,
	AccordionIcon,
	Box,
} from '@chakra-ui/react';

const propTypes = {
	dataSource: PropTypes.arrayOf(PropTypes.shape({
		title: PropTypes.oneOfType([
			PropTypes.string,
			PropTypes.node,
		]),
		onRenderItem: PropTypes.func,
	})),
	defaultIndex: PropTypes.number,
	allowMultiple: PropTypes.bool,
	allowToggle: PropTypes.bool,
	onChange: PropTypes.func,
	accordionItemProps: PropTypes.object,
	styled: PropTypes.object,
	accordionButtonStyled: PropTypes.object,
};

function Accordion({
	dataSource,
	defaultIndex,
	allowMultiple,
	allowToggle,
	onChange,
	accordionItemProps,
	styled,
	accordionButtonStyled,
}) {
	function _renderAccordionItems() {
		return dataSource.map((item, index) => (
			<AccordionItem key={index} {...accordionItemProps}>
				<Box as="h2" w="100%">
					<AccordionButton {...accordionButtonStyled}>
						<Box flex="1" textAlign="left">
							{item.title}
						</Box>
						<AccordionIcon />
					</AccordionButton>
				</Box>
				<AccordionPanel pb={4}>
					{item.onRenderItem()}
				</AccordionPanel>
			</AccordionItem>
		));
	}

	return (
		<CAccordion
			defaultIndex={defaultIndex}
			allowMultiple={allowMultiple}
			allowToggle={allowToggle}
			onChange={onChange}
			{...styled}
		>
			{_renderAccordionItems()}
		</CAccordion>
	);
}

Accordion.propTypes = propTypes;

export default Accordion;
