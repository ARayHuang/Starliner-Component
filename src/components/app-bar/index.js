import React from 'react';
import { default as MAppBar } from '@material-ui/core/AppBar';
import { default as MToolbar } from '@material-ui/core/Toolbar';
import PropTypes from 'prop-types';

const PositionType = {
	ABSOLUTE: 'absolute',
	FIXED: 'fixed',
	RELATIVE: 'relative',
	STATIC: 'static',
	STICKY: 'sticky',
};
const propTypes = {
	className: PropTypes.string,
	position: PropTypes.oneOf(Object.values(PositionType).concat('')),
	children: PropTypes.any,
};

function AppBar({ className, position, children }) {
	return (
		<MAppBar
			className={className}
			position={position}
			color="transparent"
		>
			<MToolbar>
				{children}
			</MToolbar>
		</MAppBar>
	);
}

AppBar.PositionType = PositionType;
AppBar.propTypes = propTypes;

export default AppBar;
