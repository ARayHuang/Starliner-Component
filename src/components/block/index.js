import React from 'react';
import { default as MPaper } from '@material-ui/core/Paper';
import PropTypes from 'prop-types';

const VariantType = {
	ELEVATION: 'elevation',
	OUTLINED: 'outlined',
};
const propTypes = {
	className: PropTypes.string,
	variant: PropTypes.oneOf(Object.values(VariantType).concat('')),
	shadowNumber: PropTypes.number,
	children: PropTypes.node,
	isSquare: PropTypes.bool,
};

function Block({
	className,
	variant,
	shadowNumber,
	children,
	isSquare,
}) {
	return (
		<MPaper
			className={className}
			variant={variant}
			elevation={shadowNumber}
			square={isSquare}
		>
			{children}
		</MPaper>
	);
}

Block.VariantType = VariantType;
Block.propTypes = propTypes;

export default Block;
