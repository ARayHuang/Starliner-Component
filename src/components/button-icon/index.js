import React from 'react';
import { IconButton as CIconButton, Tooltip } from '@chakra-ui/react';
import PropTypes from 'prop-types';

const VariantType = {
	SOLID: 'solid',
	OUTLINE: 'outline',
	GHOST: 'ghost',
	LIST: 'list',
};
const Size = {
	XS: 'xs',
	SM: 'sm',
	MD: 'md',
	LG: 'lg',
};
const defaultStyle = {
	border: '1px solid transparent',
	boxShadow: '0px 3px 1px -2px rgba(#000, 0.2), 0px 2px 2px 0px rgba(#000, 0.14), 0px 1px 5px 0px rgba(#000, 0.12)',
};
const propTypes = {
	className: PropTypes.string,
	variant: PropTypes.oneOf(Object.values(VariantType).concat('')),
	size: PropTypes.oneOf(Object.values(Size).concat('')),
	text: PropTypes.string,
	icon: PropTypes.element,
	onClick: PropTypes.func,
	sx: PropTypes.object,
	tooltip: PropTypes.string,
	isDisabledTooltip: PropTypes.bool,
	isDisabled: PropTypes.bool,
};
const defaultProps = {
	variant: VariantType.REGULAR,
	size: Size.SM,
	sx: {},
};

function IconButton({
	className,
	variant,
	size,
	text,
	icon,
	onClick,
	styled,
	tooltip,
	isDisabled,
	isDisabledTooltip,
}) {
	return (
		<>
			<Tooltip
				label={tooltip}
				isDisabled={isDisabledTooltip}
				borderRadius="4px">
				<CIconButton
					className={className}
					variant={variant}
					size={size}
					disabled={isDisabled}
					icon={icon}
					isRound = {true}
					onClick={onClick}
					aria-label={text}
					{...styled}
					{...defaultStyle}
				/>
			</Tooltip>
		</>
	);
}

IconButton.VariantType = VariantType;
IconButton.Size = Size;
IconButton.propTypes = propTypes;
IconButton.defaultProps = defaultProps;

export default IconButton;
