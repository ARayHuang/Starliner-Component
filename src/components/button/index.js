import React from 'react';
import { Button as CButton, Tooltip } from '@chakra-ui/react';
import PropTypes from 'prop-types';

const VariantType = {
	SOLID: 'solid',
	OUTLINE: 'outline',
	GHOST: 'ghost',
	LIST: 'list',
};
const Size = {
	XS: 'xs',
	SM: 'sm',
	MD: 'md',
	LG: 'lg',
};
const defaultStyle = {
	border: '1px solid transparent',
	borderRadius: '4px',
	boxShadow: '0px 3px 1px -2px rgba(#000, 0.2), 0px 2px 2px 0px rgba(#000, 0.14), 0px 1px 5px 0px rgba(#000, 0.12)',
};
const propTypes = {
	className: PropTypes.string,
	variant: PropTypes.oneOf(Object.values(VariantType).concat('')),
	size: PropTypes.oneOf(Object.values(Size).concat('')),
	text: PropTypes.string,
	icon: PropTypes.element,
	onClick: PropTypes.func,
	sx: PropTypes.object,
	styled: PropTypes.object,
	tooltip: PropTypes.string,
	isDisabledTooltip: PropTypes.bool,
	isDisabled: PropTypes.bool,
};
const defaultProps = {
	variant: VariantType.REGULAR,
	size: Size.SM,
	sx: {},
};

function Button({
	className,
	variant,
	size,
	text,
	icon,
	onClick,
	styled,
	sx,
	tooltip,
	isDisabled,
	isDisabledTooltip,
}) {
	return (
		<>
			<Tooltip
				label={tooltip}
				isDisabled={isDisabledTooltip}
				borderRadius="4px">
				<CButton
					className={className}
					variant={variant}
					size={size}
					disabled={isDisabled}
					leftIcon={icon}
					onClick={onClick}
					sx={{ ...defaultStyle, ...sx }}
					{...styled}
				>
					{text}
				</CButton>
			</Tooltip>
		</>
	);
}

Button.VariantType = VariantType;
Button.Size = Size;
Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;
