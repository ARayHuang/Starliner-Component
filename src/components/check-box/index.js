import Checkbox from './check-box';
import CheckboxGroup from './check-box-group';
import CheckboxGroupControl from './check-box-group-control';

export {
	Checkbox,
	CheckboxGroup,
	CheckboxGroupControl,
};
