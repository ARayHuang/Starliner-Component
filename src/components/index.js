import AppBar from './app-bar';
import Block from './block';
import { List,
	ListItem,
	ListItemLink,
	ListItemCollapse } from './list';
import SideBar from './side-bar';
import Button from './button';

export {
	AppBar,
	Block,
	List,
	ListItem,
	ListItemLink,
	ListItemCollapse,
	SideBar,
	Button,
};
