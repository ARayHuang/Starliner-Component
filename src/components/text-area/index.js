import Textarea from './textarea';
import TextareaControl from './textarea-control';

export {
	Textarea,
	TextareaControl,
};
