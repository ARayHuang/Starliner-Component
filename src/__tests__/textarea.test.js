import React from 'react';
import { shallow, mount } from 'enzyme';
import { Textarea } from '../components/text-area';


describe('textarea', () => {
	it('handle default props', () => {
		const  {
			onChange
		} = Textarea.defaultProps;

		expect(onChange).toBeInstanceOf(Function);
	});

	it('should render correctly', () => {
		const props = {
			placeholder: 'mock-placeholder',
			value: 'mock-value',
			onChange: () => {}
		};

		const wrapper = shallow(
			<Textarea {...props} />
		)

		expect(wrapper).toMatchSnapshot();
	})

	it('should mount in a full DOM', () => {
		const props = {
			placeholder: 'mock-placeholder',
			value: 'mock-value',
			onChange: () => {}
		};

		const wrapper = mount(
			<Textarea {...props} />
		);

		expect(wrapper.props().placeholder).toEqual(props.placeholder);
		expect(wrapper.props().defaultValue).toEqual(props.defaultValue);
		expect(wrapper.props().value).toEqual(props.value);
		expect(wrapper.props().onChange).toEqual(props.onChange);
	})

	it('should handle onChange event', () => {
		const onChangeMock = jest.fn();
		const value = 'mock-value';
		const event = {
			preventDefault: () => {},
			target: { value }
		}

		const wrapper = shallow(
			<Textarea value={value} onChange={onChangeMock} />
		);
		wrapper.simulate('change', event);

		expect(onChangeMock).toBeCalledWith(event);
		expect(onChangeMock).toHaveBeenCalledTimes(1);
	});
})
