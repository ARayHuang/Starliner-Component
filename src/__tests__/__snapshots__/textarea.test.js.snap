// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`textarea should render correctly 1`] = `
<Textarea
  onChange={[Function]}
  placeholder="mock-placeholder"
  value="mock-value"
/>
`;
